code/modules/research/xenobiology/vatgrowing/vatgrower.dm

```diff
@@ -170,6 +174,39 @@
 		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), get_turf(src), 'sound/effects/servostep.ogg', 100, 1), 1.5 SECONDS)
 		biological_sample.reset_sample()
 	else
-		UnregisterSignal(biological_sample, COMSIG_SAMPLE_GROWTH_COMPLETED)
+		UnregisterSignal(biological_sample, list(COMSIG_SAMPLE_GROWTH_COMPLETED, COMSIG_SAMPLE_DEPOSITED))
 		QDEL_NULL(biological_sample)
 	update_appearance()
+
+/obj/machinery/vatgrower/proc/on_sample_deposited(datum/biological_sample/sample, atom/thing)
+	SIGNAL_HANDLER
+
+/obj/machinery/vatgrower/small
+	name = "small growing vat"
+	desc = "Tastes just like the chef's soup. Fit for growing organ tissue samples."
+	icon_state = "growing_vat_small"
+	density = FALSE
+	pass_flags_self = PASSMACHINE | LETPASSTHROW
+	circuit = null
+
+	anchored_tabletop_offset = 4
+	pass_flags = PASSTABLE
+	use_plumbing = FALSE
+
+	reagent_volume = 50
+
+/obj/machinery/vatgrower/small/emag_act(mob/user, obj/item/card/emag/emag_card)
+	return
+
+/obj/machinery/vatgrower/small/on_sample_deposited(datum/biological_sample/sample, atom/thing)
+	. = ..()
+
+	if(isliving(thing))
+		var/mob/living/living_type = thing
+		if(living_type::mob_size > MOB_SIZE_SMALL)
```

Comment:
Why use `::` specifically?

Reply:
It's a type not an object. If you're meaning 'why not initial', I didn't think it mattered

Reply:
Also `thing` is a horrible name for a parameter I couldn't tell if it was a typepath or not 😛

Reply:
huh idk what I thought, these are all thinks I know